### ISSUE TITLE
Case insensitive autocomplete doesn’t work with Cyrillic letters

### ISSUE THREAD

**Issue by pepelsbey** [ohmyzsh/ohmyzsh#9947]
**Describe the bug**

Case insensitive autocomplete doesn’t work with Cyrillic letters, but works with Latin ones.

**To Reproduce**

1. Create two folders:
  - `mkdir Latin`
  - `mkdir Кириллица`
2. Type `cd l` and press Tab
3. Type `cd к` and press Tab

**Expected behavior**

- `cd Latin/` suggest on step 2
- `cd Кириллица/` suggest on step 3

**Desktop**

 - macOS
 - Latest ohmyzsh
 - zsh 5.8
 - iTerm2, Terminal

**Additional context**

- Suggest only works for Cyrillic if you explicitly type `cd К`
- My `.zshrc` is default, I’ve just installed it

**Comment by pepelsbey** [comment]
I have a feeling that it might be somehow fixed [right here](https://github.com/ohmyzsh/ohmyzsh/blob/12669f29f0843b8b980dd137f150a74511f88842/lib/completion.zsh#L18), but I’m not sure how this thing works.

```zsh
if [[ "$CASE_SENSITIVE" = true ]]; then
  zstyle ':completion:*' matcher-list 'r:|=*' 'l:|=* r:|=*'
else
```

**Comment by denysdovhan** [comment]
Please, don’t forget about Cyrillic letters from other languages, like Ї, Ґ, Є, Ў.

AFAIK, there are also specific letters in Bulgarian and Serbian languages (needs investigation)

**Comment by pepelsbey** [comment]
> Please, don’t forget about Cyrillic letters from other languages, like Ї, Ґ, Є, Ў.

Absolutely! That’s what I’m talking about. It wouldn’t make a reliable solution to list all the letters in the world. We need a universal match. I’d love to suggest it, but I’m not sure if I get what’s going on here: `'r:|=*' 'l:|=* r:|=*'`.

**Comment by acid-bong** [comment]
Sadly, not OMZ-specific, Zsh's completion matching itself is limited by ASCII[^1]:
> the matching system does not yet handle multibyte characters

And Cyrillic ofc is in the multibyte area.

[^1]: https://zsh.sourceforge.io/Doc/Release/Completion-Widgets.html#Completion-Matching-Control, right above the `m:...` block

**Comment by robbyrussell** [comment]
Hey @pepelsbey, sorry — nobody ever replied here.

I reproduced it on zsh 5.9: `ls к<TAB>` completes nothing while `ls К<TAB>` works, so you were right and it still happens.

It isn't our matcher, though. We moved to POSIX classes (`[:lower:]`/`[:upper:]`) in #11526 back in 2023, and it made no difference — zsh matches multibyte characters perfectly well but never case-folds them, so no matcher spelling fixes it. I tried your `m:{а-яё}={А-ЯЁ}` too, both as a separate list element and inlined, and neither works.

Marc gave the full answer on #13477 earlier this year: doing it by hand would mean writing out a thousand-plus character pairs, which isn't something we can carry. That's the thread to follow.

Closing this one as the same issue. Sorry it sat here so long unanswered.

🤖 Co-authored with Claude Code
